files/acl/restrict-unpublished-files.php

```diff
@@ -105,3 +104,43 @@ function get_attachment_id_from_file_path( $path ) {
 
 	return $attachment_id;
 }
+
+/**
+ * PURGE all attachments for a given post.
+ *
+ * This is to allow us to propagate visibility changes ASAP to the edges
+ * and also to allow us to cache public files for much longer (which comes
+ * some performance benefits).
+ *
+ * @param array urls Array of URLs to be purged
+ * @param int post_id The ID of the post for which we're purging URLs
+ *
+ * @return array An array of URLs to be purged
+ */
+function purge_attachments_for_post( $urls, $post_id ) {
+	$post = get_post( $post_id );
+
+	if ( empty( $post ) ) {
+		return $urls;
+	}
+
+	$attachment_ids = get_posts( [
+		'post_parent'    => $post->ID,
+		'post_type'      => 'attachment',
+		'posts_per_page' => 250, // Set a reasonably high limit (instead of -1 as default)
+		'orderby' => 'ID', // For performance (instead of `date` as default)
+		'order' => 'ASC',
+		'fields' => 'ids',
+		'suppress_filters' => false,
```

Comment:
@mjangda I'm kind of conflicted on this one being set to false because we'll be at risk of userland code interfering with our logic here. I think it's ok to potentially sacrifice a tiny bit of performance (I'm assuming this is set to false for APC to be able to cache the query, but, realistically, this particular query would only run in this given context that also won't happen on the frontend (🤞 )
```suggestion
		'suppress_filters' => true,
```

Reply:
Yep, that seems reasonable.